### PR TITLE
Properly format en and em dashes in reference titles

### LINKED ISSUE #11: Convert common TeX to unicode
I set it up [here](https://clima.github.io/TimeMachine.jl/dev/references/), and noticed a few TeX artifacts.

I would suggest at least converting two dashes (`--`) to an en-dash (`–`) as those are very common and hard to type.

### PR DESCRIPTION
**Pull request by ali-ramadhan**
This PR starts adding support for converting common TeX to unicode starting with en and em dashes.

This PR depends on #13.

Resolves #11 (?)

### CODE REVIEW

**Comment by ali-ramadhan** [comment]
It's a very naive approach that scales as O(B*R) where B is the number of items in the bibliography and R is the number of possible replacements.

But maybe we don't expect too many replacements. Certainly there aren't that many text replacements we want to do for titles I think.

File: src/bibliography.jl [hunk 1]
```diff
@@ -1,9 +1,20 @@
...
+const tex2unicode_replacements = (
+    "--"  => "–",
+    "---" => "—"
+)
+
+function tex2unicode(s)
+    for replacement in tex2unicode_replacements
+        s = replace(s, replacement)
+    end
+    return s
```
Comment (simonbyrne) [inline src/bibliography.jl:15]: I think you can use multiple pairs in one `replace`
```suggestion
        replace(s, tex2unicode_replacements...)
```

Comment (ali-ramadhan) [reply src/bibliography.jl:15]: Woah nice!

Comment (ali-ramadhan) [reply src/bibliography.jl]: Ah looks like

```julia
replace(A, old_new::Pair...; [count::Integer])
```

is defined in general but for strings we are stuck with

```julia
replace(s::AbstractString, pat=>r; [count::Integer])
```

:(

**Review by charleskawczynski — COMMENTED** [review]
Seems fine by me. Rebase/squash first?

File: src/bibliography.jl (hunk 1 above)
Comment (charleskawczynski) [reply src/bibliography.jl:15]: Ah, ok.

**Review by charleskawczynski — APPROVED** [review]

File: src/bibliography.jl (hunk 1 above)
Comment (simonbyrne) [reply src/bibliography.jl:15]: Ah, I just realised. Upstream issue https://github.com/JuliaLang/julia/issues/35327

**Pull request merged**